draft-ietf-quic-transport.md

```diff
@@ -831,29 +768,46 @@ larger resource commitments at the receiver.  Thus there is a trade-off between
 resource commitment and overhead when determining how large a limit is
 advertised.
 
-A receiver MAY use an autotuning mechanism to tune the frequency and amount that
-it increases data limits based on a round-trip time estimate and the rate at
-which the receiving application consumes data, similar to common TCP
-implementations.
+A receiver MAY use an autotuning mechanism to tune the frequency and amount of
+advertised additional credit.
 
-If a sender runs out of flow control credit, it will be unable to send new
-data. That is, the sender is blocked. A blocked sender SHOULD send a
-STREAM_DATA_BLOCKED or DATA_BLOCKED frame.  A receiver uses these frames for
-debugging purposes.  A receiver MUST NOT wait for a STREAM_DATA_BLOCKED or
-DATA_BLOCKED frame before sending MAX_STREAM_DATA or MAX_DATA, since doing so
-will mean that a sender will be blocked for an entire round trip and the peer
-might never send a STREAM_DATA_BLOCKED or DATA_BLOCKED frame.
+If a sender runs out of flow control credit, it will be unable to send new data
+and is considered blocked.  It is generally considered best to not let the
+sender go into quiescence.  To avoid blocking a sender, and to reasonably
+account for the possibility of loss, a receiver should send a MAX_DATA or
+MAX_STREAM_DATA frame at least two round trips before it expects the sender to
+get blocked.
 
-It is generally considered best to not let the sender go into quiescence if
-avoidable.  To avoid blocking a sender, and to reasonably account for the
-possibility of loss, a receiver should send a MAX_DATA or MAX_STREAM_DATA frame
-at least two round trips before it expects the sender to get blocked.
+A receiver MUST NOT wait for a STREAM_DATA_BLOCKED or DATA_BLOCKED frame before
+sending MAX_STREAM_DATA or MAX_DATA, since doing so will mean that a sender will
+be blocked for at least an entire round trip, and potentially for longer if the
+peer chooses to not send STREAM_DATA_BLOCKED or DATA_BLOCKED frames.
 
-A sender sends a single STREAM_DATA_BLOCKED or DATA_BLOCKED frame only once when
-it reaches a data limit.  A sender SHOULD NOT send multiple STREAM_DATA_BLOCKED
-or DATA_BLOCKED frames for the same data limit, unless the original frame is
-determined to be lost.  Another STREAM_DATA_BLOCKED or DATA_BLOCKED frame can be
-sent after the data limit is increased.
+
+## Handling Stream Cancellation {#stream-cancellation}
+
+Given enough time, both endpoints must agree on flow control state, to avoid
```

Comment:
Endpoints need to eventually agree on the amount of flow control credit that has been consumed, to avoid either exceeding limits or deadlocking.